share/spack/setup-env.sh

```diff
@@ -340,17 +340,38 @@ else
     eval `spack --print-shell-vars sh`
 fi;
 
+_sp_get_generic_arch() {
+    local plat os target 
+    target=`spack arch --target`
+    os=`spack arch --operating-system`
+    plat=`spack arch --platform`
+    spack arch --known-targets | 
```

Comment:
All of these calls to `spack arch` are very slow. This will likely cause problems for people running Spack on an NFS filesystem.

Reply:
This is what `spack --print-shell-vars` is meant for -- it's supposed to give you everything you need in one invocation, and it happens very early in `spack`'s main method, so you don't have to load a bunch of modules.

So new info for `setup-env.sh` should go through that.